spec.html

```diff
@@ -29851,11 +29851,23 @@ <h1>Runtime Semantics: GetSubstitution ( _matched_, _str_, _position_, _captures
             1. Assert: _position_ &le; _stringLength_.
             1. Assert: _captures_ is a possibly empty List of Strings.
             1. Assert: Type(_replacement_) is String.
+            1. Let _replacementLength_ be the number of code units in _replacement_.
             1. Let _tailPos_ be _position_ + _matchLength_.
             1. Let _m_ be the number of elements in _captures_.
             1. If _namedCaptures_ is not *undefined*, then
               1. Set _namedCaptures_ to ? ToObject(_namedCaptures_).
-            1. Let _result_ be the String value derived from _replacement_ by copying code unit elements from _replacement_ to _result_ while performing replacements as specified in <emu-xref href="#table-45"></emu-xref>. These `$` replacements are done left-to-right, and, once such a replacement is performed, the new replacement text is not subject to further replacements.
+            1. Let _result_ be the empty String.
+            1. Let _k_ be 0.
+            1. Repeat, while _k_ &lt; _replacementLength_
+              1. Let _replaceablePattern_ be the longest sequence of consecutive code units from _replacement_ beginning with the code unit at index _k_ such that _replaceablePattern_ matches the &ldquo;Code units&rdquo; column of a row in <emu-xref href="#table-45"></emu-xref>.
+              1. If _replaceablePattern_ is not empty, then
+                1. Let _patternReplacement_ be the result of the &ldquo;Replacement text&rdquo; column of the corresponding row.
```

Comment:
There should be a `ReturnIfAbrupt(_patternReplacement_ )` following this step, I think, since the named captures case can throw.

(Or that row - or all rows - could be factored out from the table into explicit algorithm steps here.)

Reply:
Nice catch! Updated and rebased.